.github/workflows/build.yml

```diff
@@ -0,0 +1,38 @@
+name: build
+on:
+  release:
+    types: [created]
+  push:
+    branches:
+      - "**"
+    paths:
+      - "header-gen/**"
+      - "**/build.yml"
```

Comment:
We have a different workflow that already builds and runs the tests on push.

We could either merge the two workflows into one, or remove this here (and maybe add the paths entry to the `header-gen.yml` file).